packages/kamino-sdk/src/Kamino.ts

```diff
@@ -5071,33 +5072,97 @@ export class Kamino {
+      mintInfo.mintAuthority = null;
+    } else {
+      mintInfo.mintAuthority = new PublicKey(mintInfo.mintAuthority);
+    }
+
+    mintInfo.supply = u64.fromBuffer(mintInfo.supply);
+    mintInfo.isInitialized = mintInfo.isInitialized !== 0;
+
+    if (mintInfo.freezeAuthorityOption === 0) {
+      mintInfo.freezeAuthority = null;
+    } else {
+      mintInfo.freezeAuthority = new PublicKey(mintInfo.freezeAuthority);
+    }
+
+    return mintInfo;
+  }
+
   /**
    * Get a list of user's Kamino strategy positions
    * @param wallet user wallet address
-   * @param strategyFilters
    * @returns list of kamino strategy positions
    */
-  getUserPositions = async (
-    wallet: PublicKey,
-    strategyFilters: StrategiesFilters = { strategyCreationStatus: 'LIVE' }
-  ): Promise<KaminoPosition[]> => {
-    const userTokenAccounts = await this.getAllTokenAccounts(wallet);
-    const liveStrategies = await this.getAllStrategiesWithFilters({ ...strategyFilters, owner: wallet });
-    const positions: KaminoPosition[] = [];
-    for (const tokenAccount of userTokenAccounts) {
-      const accountData = tokenAccount.account.data as Data;
-      const strategy = liveStrategies.find(
-        (x) => x.strategy.sharesMint.toString() === accountData.parsed.info.mint.toString()
+  getUserPositions = async (wallet: PublicKey): Promise<KaminoPosition[]> => {
+    const accounts = await this._connection.getParsedTokenAccountsByOwner(wallet, {
+      programId: TOKEN_PROGRAM_ID,
+    });
+
+    const mints = accounts.value.map((accountInfo) => {
+      return new PublicKey(accountInfo.account.data.parsed.info.mint);
+    });
+    const mintInfos = await batchFetch(mints, (chunk) => this.getConnection().getMultipleAccountsInfo(chunk));
+
+    const kaminoMintWithAccountInfo: [PublicKey, AccountInfo<ParsedAccountData>][] = [];
+
+    for (const index of mints.keys()) {
+      const mint = mints[index];
+      const mintInfo = mintInfos[index];
+      const accountInfo = accounts.value[index];
+
+      if (!mint || !mintInfo || !accountInfo) continue;
+
+      const [expectedMintAuthority] = PublicKey.findProgramAddressSync(
+        [Buffer.from('authority'), mint.toBuffer()],
+        this.getProgramID()
       );
-      if (strategy) {
-        positions.push({
-          shareMint: strategy.strategy.sharesMint,
-          strategy: strategy.address,
-          sharesAmount: new Decimal(accountData.parsed.info.tokenAmount.uiAmountString),
-          strategyDex: numberToDex(strategy.strategy.strategyDex.toNumber()),
-        });
+
+      const mintData = this._deserializeMint(mintInfo.data);
+
+      if (mintData.mintAuthority !== null && mintData.mintAuthority.equals(expectedMintAuthority)) {
+        kaminoMintWithAccountInfo.push([mint, accountInfo.account]);
       }
     }
+
+    const strategies = await batchFetch(
+      kaminoMintWithAccountInfo,
+      (chunk) => Promise.all(chunk.map(([mint]) => this.getStrategyByKTokenMint(mint))),
```

Comment:
This needs to make a get multiple accounts as well, dont Promise.all multiple requests. We have existing methods that fetch multiple strategies with batching already in the SDK

Reply:
updated